### PR TITLE
Dropped zaudio and added a html5 audio view to the file plugin

### LINKED ISSUE #5928: Drop zaudio in favor of HTML5 <audio> + shim
If I'm not mistaken, browsers should pretty much be able to play mp3's themselves these days without a flash fallback.

Also feels like this integration belongs with the "file" plugin.

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #5928
- [x] add upgrade docs about the removal of this plugin
- [x] move code to seperate repo

### CODE REVIEW

**Comment by juho-jaakkola** [comment]
Removing a plugin completely should definitely be mentioned in the changelog. It's probably best to call it a feature.

**Comment by jdalsem** [comment]
> Removing a plugin completely should definitely be mentioned in the changelog. It's probably best to call it a feature.

hmm don't like it being a feature :( but there indeed is not a better option AFAICS

**Comment by jdalsem** [comment]
changed it to a feature and also removed a reference in the docs

File: mod/file/views/default/file/specialcontent/audio/default.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
+/**
+ * Outputs an audio player in full view
+ *
+ * @uses $vars['entity']
+ */
+
+$file = $vars['entity'];
```
Comment (jeabakker) [inline mod/file/views/default/file/specialcontent/audio/default.php]: please use elgg_extract

File: mod/file/views/default/file/specialcontent/audio/default.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+if (!$vars['full_view']) {
```
Comment (jeabakker) [inline mod/file/views/default/file/specialcontent/audio/default.php]: please use elgg_extract

**Comment by hypeJunction** [comment]
IMHO, should create Elgg/zaudio repo just in case

File: mod/file/views/default/file/specialcontent/audio/default.php (hunk 1 above)
Comment (jeabakker) [inline mod/file/views/default/file/specialcontent/audio/default.php]: and make sure you have an entity

**Comment by jdalsem** [comment]
> IMHO, should create Elgg/zaudio repo just in case

yeah, but i can not do that :)

**Comment by juho-jaakkola** [comment]
See also https://community.elgg.org/discussion/view/2109099/should-we-add-a-remove-commit-message-type

**Comment by brettp** [comment]
https://github.com/Elgg/zaudio

**Comment by jdalsem** [comment]
can somebody with a bit more git skills move the current code + history of the zaudio plugin to that repo?

**Comment by ewinslow** [comment]
@jdalsem https://docs.google.com/a/elgg.org/document/d/1vqZ9M9bjoCThscxMh-SUgJ8wJlbwpNEim-lcrGIk2zU/edit#heading=h.tty8d3mtk9d7

**Comment by ewinslow** [comment]
@jdalsem Still going to make this happen? Would be great to get rid of zaudio...

**Comment by jdalsem** [comment]
Will make this happen this week

**Comment by jdalsem** [comment]
The plugin has been moved to its own repo (don't know who did it) https://github.com/Elgg/zaudio and i update the commit to include the BREAKING CHANGE description

**Comment by jdalsem** [comment]
hmmm need to rebase this

**Comment by mrclay** [comment]
I'd like to be able to turn the audio player off, considering Elgg streaming is still super inefficient.

**Comment by jdalsem** [comment]
you could override the view....

**Comment by ewinslow** [comment]
So my plan with the other plugins was to remove from a very low version of Elgg like 1.10 and include via composer instead. That way it's not even a breaking change, really.

Also that way we won't accidentally do any global refactoring on elgg/elgg that affects the zaudio plugin in this repo, forcing us to the merge those commits.

On thing we'd need to do is tag the 1.x branch in Elgg/zaudio repo so we don't pull any breaking changes into 1.x

Would it be too much to do that in this case?

Thanks for doing this! Sorry to punish a good deed...

**Comment by mrclay** [comment]
What's left on this?

**Comment by ewinslow** [comment]
Rebase?

**Comment by ewinslow** [comment]
@jdalsem Can you rebase this? It is blocking 2.0

**Comment by ewinslow** [comment]
#8821

**Pull request closed without merging**